functions/src/lobby.ts

```diff
@@ -90,13 +91,7 @@ export const startGame = functions.https.onCall(async (data: unknown, context):
       throw new functions.https.HttpsError("permission-denied", "Not the host of the game!");
     }
 
-    const privatePlayerCollection = getPrivatePlayerCollection(lobby.ref);
-    for (const uid of uids) {
-      privatePlayerCollection.doc(uid).create({ role: "CAT" });
-    }
-
-    // await lobbyCollection.doc(data.code).update({ state: "PROMPT" });
-    await startPrompt(lobby, transaction);
+    await assignRole(lobby.ref);
```

Comment:
this function times out because you're using a transaction within a transaction. So it never goes to the ROLE state